Framework/Geometry/test/ComponentInfoTest.h

```diff
@@ -649,24 +649,38 @@ class ComponentInfoTest : public CxxTest::TestSuite {
                      detectorPos);
   }
 
-  void test_StructuredPanel_for_single_rectangular_bank() {
+  void test_throws_if_ComponentType_is_not_Quadrilateral() {
     auto instrument =
         ComponentCreationHelper::createTestInstrumentRectangular2(1, 4);
     auto wrappers = InstrumentVisitor::makeWrappers(*instrument);
     const auto &componentInfo = std::get<0>(wrappers);
 
-    // find structured panel
-    size_t structuredIndex = 0;
-    for (size_t i = componentInfo->root(); i > 0; --i) {
-      if (componentInfo->componentType(i) ==
-              Mantid::Beamline::ComponentType::Structured ||
-          componentInfo->componentType(i) ==
-              Mantid::Beamline::ComponentType::Rectangular) {
-        structuredIndex = i;
-        break;
-      }
-    }
+    // find quadrilateral component
+    size_t structuredIndex = componentInfo->root() - 3;
+    // Does not throw for valid index
+    TS_ASSERT_THROWS_NOTHING(
+        componentInfo->quadrilateralComponent(structuredIndex));
+    // Throws for other non quadrilateral component
+    TS_ASSERT_THROWS(
```

Comment:
very minor point, catch as `std::runtime_error&` not `std::runtime_error` some compilers complain about this.